### PR TITLE
Deploy new buildimages with default configuration for curl, wget and awscli

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Update build images used in the CI with new version having default retry enabled.
Also contains minor fix on invoke task for buildimage update

### Motivation
Increate CI stability

### Additional Notes
Tests done from buildimages PR
https://github.com/DataDog/datadog-agent-buildimages/pull/464
https://github.com/DataDog/datadog-agent-builders/pull/243

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/unit-tests/testdata/fake_gitlab-ci.yml
Comment (chouetz) [inline tasks/unit-tests/testdata/fake_gitlab-ci.yml:1]: No need to review this file, it's only used for tests

File: tasks/pipeline.py [hunk 1]
```diff
@@ -860,7 +860,8 @@ def update_gitlab_config(file_path, image_tag, test_version):
     with open(file_path, "r") as gl:
         file_content = gl.readlines()
     gitlab_ci = yaml.safe_load("".join(file_content))
-    suffixes = [name for name in gitlab_ci["variables"].keys() if name.endswith("SUFFIX")]
+    # TEST_INFRA_DEFINITION_BUILDIMAGE label format differs from other buildimages
+    suffixes = [name for name in gitlab_ci["variables"].keys() if name.endswith("SUFFIX") and not name.startswith("TEST_INFRA_DEFINITION")]
```
Comment (amenasria) [inline tasks/pipeline.py:864]:
```suggestion
    suffixes = [name for name in gitlab_ci["variables"] if name.endswith("SUFFIX") and not name.startswith("TEST_INFRA_DEFINITION")]
```

**💬 suggestion:** ‏More pythonic way of doing this, wdyt?

File: tasks/pipeline.py [hunk 2]
```diff
@@ -902,9 +903,11 @@ def trigger_build(ctx, branch_name=None, create_branch=False):
     """
     if create_branch:
         ctx.run(f"git checkout -b {branch_name}")
-    ctx.run("git add .gitlab-ci.yml .circleci/config.yml")
     answer = input("Do you want to trigger a pipeline (will also commit and push)? [Y/n]\n")
     if len(answer) == 0 or answer.casefold() == "y":
+        ctx.run("git add .gitlab-ci.yml .circleci/config.yml")
```
Comment (amenasria) [inline tasks/pipeline.py:908]: **❓ question:** ‏Why did you put the `git add` in the loop ?

File: tasks/pipeline.py [hunk 3]
```diff
@@ -902,9 +903,11 @@ def trigger_build(ctx, branch_name=None, create_branch=False):
...
         ctx.run("git commit -m 'Update buildimages version'")
         ctx.run(f"git push origin {branch_name}")
+        # Let Gitlab create the first events before triggering a new pipeline
+        time.sleep(30)
```
Comment (amenasria) [inline tasks/pipeline.py:912]: **💬 suggestion:** ‏Wow that's a lot ! I think we should also print a message saying smth like "Waiting 30s for Gitlab to create the first events..." to have a better developer experience. WDYT?

Comment (amenasria) [inline tasks/pipeline.py:912]:
```suggestion
        # Let Gitlab create the first events before triggering a new pipeline
        waiting_time = 30
        print(f"Waiting {waiting_time}s for Gitlab to create the first events...")
        time.sleep(waiting_time)
```

**Review by amenasria — APPROVED** [review]
Few comments, otherwise LGTM !

File: tasks/pipeline.py (hunk 2 above)
Comment (chouetz) [reply tasks/pipeline.py:908]: To still be able to do a `git diff` to view the changes

File: tasks/pipeline.py (hunk 3 above)
Comment (chouetz) [reply tasks/pipeline.py]: I decreased to 10s and printed a message

**Review by amenasria — APPROVED** [review]
LGTM !

**Pull request merged**